### DISCUSSION TITLE
Which is the best way to apply a filter on the projection?

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#501] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v12.4.0

### Description

Hi,

I need to avoid storing projections that match a specific filter in order to reduce the size of the projetion collection.
Which is the recommended way of doing it with Mia-Platform Fast Data?
I am using the projection storer v1.2.2.

I have thought about implementing a custom adapter but it seems a bit overkill since I am using Debezium and I would like to use the adapter that is provided out-of-the-box.

Thanks!

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @Lucascanna 

The custom message adapter would not solve the issue: even if it returns `null` or `undefined`, both PS and RTU will try to write a record on Mongodb.

@albertotessarotto we can think in the future to add the possibility to perform a filtering at the CDC level.

If you have access to Debezium, however, you can try using [message filtering](https://debezium.io/documentation/reference/stable/transformations/filtering.html). Have you considered that for your use case?

**Reply by lucascanna** [reply]
Thanks for the suggestion @flower-of-the-bridges
I will try with the Debezium message filtering.
